xcodeproj/internal/xcode_targets.bzl

```diff
@@ -285,17 +285,22 @@ def _merge_xcode_target(*, src, dest):
         build_settings,
     )
 
+    product = _merge_xcode_target_product(
+        src = src.product,
+        dest = dest.product,
+    )
+
+    if build_settings.get("PRODUCT_MODULE_NAME", None) == product.name:
+        build_settings.pop("PRODUCT_MODULE_NAME")
```

Comment:
This is just choosing to rely on the Xcode default for this value, correct? Is it possible to move this _just_ into the generation of `src._build_settings` or `dest._build_settings` instead to just never set it if it's the default behavior (this might be what the other file changes actually are, not sure)?

Reply:
This accounts for target merging, where until we merge them together the product name wouldn't match the module name. And yes, this (whole PR) is relying on the fact that `PRODUCT_MODULE_NAME = $(PRODUCT_NAME:c99extidentifier)`. And we already don't set it at the library and top_level when they are equal, but this catches the case where target merging would have them be equal.

Reply:
Now that I think about that, we might run into an issue where because of target merging we _do_ need `PRODUCT_MODULE_NAME` set, because it is _now_ different. 🤔